nixos/modules/services/continuous-integration/buildkite-agents.nix

```diff
@@ -43,10 +44,20 @@ in
         type = types.package;
       };
 
-      dataDir = mkOption {
-        default = "/var/lib/buildkite-agent";
-        description = "The workdir for the agent";
-        type = types.str;
+      userName = mkOption {
+        readOnly = true;
+        default = name;
+        description = ''
+          Username of the systemd service this will run as.
```

Comment:
Maybe use `default = "buildkite-${name}"` instead? I think it would be weird for it to e.g. allocate a user named `foo` just because one uses `foo` as the attribute name.